Night;y CI failing with latest pylint

pylint released a new version on Feb 25th and since then CI has been failing the nightly runs. Here is the log for the pylint checks

```
************* Module qiskit_nature.second_q.operators.spin_op
Warning: qiskit_nature/second_q/operators/spin_op.py:278:20: R1731: Consider using 'max_index = max(max_index, index)' instead of unnecessary if block (consider-using-max-builtin)
************* Module qiskit_nature.second_q.operators.fermionic_op
Warning: qiskit_nature/second_q/operators/fermionic_op.py:219:20: R1731: Consider using 'max_index = max(max_index, index)' instead of unnecessary if block (consider-using-max-builtin)
************* Module qiskit_nature.second_q.operators.bosonic_op
Warning: qiskit_nature/second_q/operators/bosonic_op.py:217:20: R1731: Consider using 'max_index = max(max_index, index)' instead of unnecessary if block (consider-using-max-builtin)
************* Module qiskit_nature.second_q.problems.properties_container
Warning: qiskit_nature/second_q/problems/properties_container.py:83:8: R1737: Use 'yield from' directly instead of yielding each element one by one (use-yield-from)

------------------------------------
Your code has been rated at 10.00/10

make: *** [Makefile:43: lint] Error 8
Error: Process completed with exit code 2.
```

--- a/qiskit_nature/second_q/operators/bosonic_op.py
+++ b/qiskit_nature/second_q/operators/bosonic_op.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2023.
+# (C) Copyright IBM 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit_nature/second_q/operators/bosonic_op.py
+++ b/qiskit_nature/second_q/operators/bosonic_op.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2024.
+# (C) Copyright IBM 2023, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit_nature/second_q/operators/bosonic_op.py
+++ b/qiskit_nature/second_q/operators/bosonic_op.py
@@ -214,8 +214,7 @@ class BosonicOp(SparseLabelOp):
             for term in key.split():
                 index = int(term[2:])
                 if num_so is None:
-                    if index > max_index:
-                        max_index = index
+                    max_index = max(max_index, index)
                 elif index >= num_so:
                     raise QiskitNatureError(
                         f"The index, {index}, from the label, {key}, exceeds the number of spin "

--- a/qiskit_nature/second_q/operators/fermionic_op.py
+++ b/qiskit_nature/second_q/operators/fermionic_op.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit_nature/second_q/operators/fermionic_op.py
+++ b/qiskit_nature/second_q/operators/fermionic_op.py
@@ -216,8 +216,7 @@ class FermionicOp(SparseLabelOp):
             for term in key.split():
                 index = int(term[2:])
                 if num_so is None:
-                    if index > max_index:
-                        max_index = index
+                    max_index = max(max_index, index)
                 elif index >= num_so:
                     raise QiskitNatureError(
                         f"The index, {index}, from the label, {key}, exceeds the number of spin "

--- a/qiskit_nature/second_q/operators/spin_op.py
+++ b/qiskit_nature/second_q/operators/spin_op.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit_nature/second_q/operators/spin_op.py
+++ b/qiskit_nature/second_q/operators/spin_op.py
@@ -275,8 +275,7 @@ class SpinOp(SparseLabelOp):
                 # sub_terms[0] is the base, sub_terms[1] is the (optional) exponent
                 index = int(sub_terms[0][2:])
                 if num_s is None:
-                    if index > max_index:
-                        max_index = index
+                    max_index = max(max_index, index)
                 elif index >= num_s:
                     raise QiskitNatureError(
                         f"The index, {index}, from the label, {key}, exceeds the number of "

--- a/qiskit_nature/second_q/problems/properties_container.py
+++ b/qiskit_nature/second_q/problems/properties_container.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2022, 2023.
+# (C) Copyright IBM 2022, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit_nature/second_q/problems/properties_container.py
+++ b/qiskit_nature/second_q/problems/properties_container.py
@@ -80,8 +80,7 @@ class PropertiesContainer(MutableSet):
         return len(self._properties)
 
     def __iter__(self) -> Generator[SparseLabelOpsFactory, None, None]:
-        for prop in self._properties.values():
-            yield prop
+        yield from self._properties.values()
 
     def _getter(self, _type: type) -> SparseLabelOpsFactory | None:
         """An internal utility method to handle the attribute getter implementation.

